lib/spack/env/cc

```diff
@@ -108,6 +108,12 @@ case "$command" in
         ;;
     ld)
         mode=ld
+        for arg in "$@"; do
+            if [ "$arg" = -S ]; then
```

Comment:
The output was from an earlier modification I made and then abandoned because it did not fix the problem.